### PR TITLE
Fix filenaming

### PR DESCRIPTION
**Pull request by jdegerickx**
- temporal dimension is stripped from classification result right after prediction UDF
- postprocessing no longer needs temporal dimension as input
- to have correct filenames, I needed temporal_extent as additional input to _cropland_map and _croptype_map functions
- "period" is now added as an attribute to each WorldCerealProduct

All changes were tested both for cropland and crop type and everything looked fine (both names of resulting files + visualization)

### CODE REVIEW

File: src/worldcereal/openeo/mapping.py [hunk 1]
```diff
@@ -21,6 +22,7 @@
 
 def _cropland_map(
     inputs: DataCube,
+    temporal_extent: TemporalContext,
```
Comment (kvantricht) [inline src/worldcereal/openeo/mapping.py:25]: The docstring should be updated to cover this additional required argument

File: src/worldcereal/job.py
Comment (kvantricht) [inline src/worldcereal/job.py:1]: To be honest, I don't really understand the need for this `period` attribute. I would instead use the original `TemporalContext` parameter as attribute `temporal_extent` for the product. It is exactly meant for this purpose without the need to define an additional string-based attribute.

File: src/worldcereal/job.py [hunk 2]
```diff
@@ -135,6 +139,10 @@ def generate_map(
         # Make a connection to the OpenEO backend
         connection = BACKEND_CONNECTIONS[backend_context.backend]()
 
+    # Get start and end dates for product file naming
+    start = temporal_extent.start_date.replace("-", "")
```
Comment (kvantricht) [inline src/worldcereal/job.py:143]: Why do the dashes have to go? Is that more readable? These two lines - if we go for the use of `temporal_extent` - are not needed. Can be plugged directly into `filename_prefix` constructor.

File: src/worldcereal/job.py [hunk 3]
```diff
@@ -164,27 +173,29 @@ def generate_map(
...
+            postprocess_parameters=postprocess_parameters,
+        )
 
         # Save final mask if required
         if croptype_parameters.save_mask:
             cropland_mask = cropland_mask.save_result(
                 format="GTiff",
                 options=dict(
-                    filename_prefix=f"{WorldCerealProductType.CROPLAND.value}",
+                    filename_prefix=f"{WorldCerealProductType.CROPLAND.value}_{start}_{end}",
```
Comment (kvantricht) [inline src/worldcereal/job.py:188]: As mentioned above, can be directly taken from `temporal_extent`

File: src/worldcereal/job.py [hunk 4]
```diff
@@ -203,12 +214,13 @@ def generate_map(
         JOB_OPTIONS.update(job_options)
 
     # Execute the job
+    filename_output = f"{product_type.value}_{start}_{end}"
```
Comment (kvantricht) [inline src/worldcereal/job.py:217]: Is a separate line needed for this? A bit inconsistent with the `cropland_mask` filename construction.

File: src/worldcereal/job.py [hunk 5]
```diff
@@ -241,6 +253,7 @@ def generate_map(
         products[asset_name] = {
             "url": asset.href,
             "type": asset_type,
+            "period": f"{start}_{end}",
```
Comment (kvantricht) [inline src/worldcereal/job.py:256]: I suggest to change this to: `"temporal_extent": temporal_extent`

File: src/worldcereal/openeo/mapping.py [hunk 6]
```diff
@@ -80,13 +82,18 @@ def _cropland_map(
         ],
     )
 
+    # Get rid of temporal dimension
+    classes = classes.reduce_dimension(dimension="t", reducer="mean")
+
     # Postprocess
     if postprocess_parameters.enable:
         if postprocess_parameters.save_intermediate:
+            start = temporal_extent.start_date.replace("-", "")
```
Comment (kvantricht) [inline src/worldcereal/openeo/mapping.py:91]: see earlier comment on dashes and the need yes/no to put this in separate line.

File: src/worldcereal/openeo/mapping.py [hunk 7]
```diff
@@ -160,17 +170,22 @@ def _croptype_map(
...
+    classes = classes.reduce_dimension(dimension="t", reducer="mean")
+
     # Mask cropland
     if cropland_mask is not None:
         classes = classes.mask(cropland_mask == 0, replacement=254)
 
     # Postprocess
     if postprocess_parameters.enable:
         if postprocess_parameters.save_intermediate:
+            start = temporal_extent.start_date.replace("-", "")
```
Comment (kvantricht) [inline src/worldcereal/openeo/mapping.py:183]: Idem

**Review by kvantricht — CHANGES_REQUESTED** [review]
My most important remark is on the introduction of `period` which I think is not needed if we stick to the already defined `TemporalContext`

**Comment by jdegerickx** [comment]
Issues addressed. 
Tested for cropland, everything looks fine!

**Review by kvantricht — APPROVED** [review]
Looks good!

**Pull request merged**